packages/@react-aria/spinbutton/src/useSpinButton.ts

```diff
@@ -137,45 +137,37 @@ export function useSpinButton(
     }
   }, [textValue]);
 
-  const onIncrementPressStart = useCallback(
-    (initialStepDelay: number) => {
-      clearAsync();
-      propsRef.current.onIncrement();
-      // Start spinning after initial delay
-      _async.current = window.setTimeout(
-        () => {
-          if (isNaN(maxValue) || isNaN(value) || value < maxValue) {
-            onIncrementPressStart(60);
-          }
-        },
-        initialStepDelay
-      );
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [onIncrement, maxValue, value]
-  );
-
-  const onDecrementPressStart = useCallback(
-    (initialStepDelay: number) => {
-      clearAsync();
-      propsRef.current.onDecrement();
-      // Start spinning after initial delay
-      _async.current = window.setTimeout(
-        () => {
-          if (isNaN(minValue) || isNaN(value) || value > minValue) {
-            onDecrementPressStart(60);
-          }
-        },
-        initialStepDelay
-      );
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [onDecrement, minValue, value]
-  );
+  let onIncrementPressStart = useEffectEvent((initialStepDelay: number) => {
+    clearAsync();
+    onIncrement();
+    // Start spinning after initial delay
+    _async.current = window.setTimeout(
+      () => {
+        if (isNaN(maxValue) || isNaN(value) || value < maxValue) {
+          onIncrementPressStart(60);
+        }
+      },
+      initialStepDelay
+    );
+  });
+
+  let onDecrementPressStart = useEffectEvent((initialStepDelay: number) => {
+    clearAsync();
+    onDecrement();
+    // Start spinning after initial delay
+    _async.current = window.setTimeout(
+      () => {
+        if (isNaN(minValue) || isNaN(value) || value > minValue) {
+          onDecrementPressStart(60);
+        }
+      },
+      initialStepDelay
+    );
+  });
```

Comment:
I think the only changes we needed in this file were converting these to `useEffectEvent` (in #4564). Do you remember why you made all the other changes?

Reply:
got carried away optimizing it would look like, I think your assessment is correct.